### PR TITLE
feat: add completion spec for tmuxinator

### PR DESCRIPTION
**Pull request by christianmeier**
added spec for tmuxinator

### CODE REVIEW

**Comment by christianmeier** [comment]
👍

File: src/tmuxinator.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        name: "PROJECT",
+        description: "Project name 1",
+        isOptional: true,
+      },
+      {
+        name: "ARGS",
+        description: "Argument",
+        isOptional: true,
+      },
+      ],
```
Comment (fedeci) [inline src/tmuxinator.ts:121]: There are some options missing for the start command https://github.com/tmuxinator/tmuxinator#starting-a-session, could you add them please?

File: src/tmuxinator.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      description:
+        "Lists commands available in tmuxinator",
+    },
+    {
+      name: "completions",
+      description:
+        "Used for shell completion",
+    },
+    {
+      name: "copy",
```
Comment (fedeci) [inline src/tmuxinator.ts:16]: There are other subcommands missing aliases, could you add them?
```suggestion
      name: ["copy", "c", "cp"],
```

File: src/tmuxinator.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        name: "EXISTING",
+        description: "The existing project",
+        isOptional: true,
+      },
+      {
+        name: "NEW",
+        description: "Name new project",
+        isOptional: true,
+      },
+      ],
```
Comment (fedeci) [inline src/tmuxinator.ts:29]: I am not sure whether these are optional 🤔, what happens if you do not provide them?

File: src/tmuxinator.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        name: "PROJECT1",
+        description: "Project name 1",
+        isOptional: true,
+      },
+      {
+        name: "PROJECT1",
+        description: "Project name 1",
+        isOptional: true,
+      },
+      ],
```
Comment (fedeci) [inline src/tmuxinator.ts:61]: Why this takes two identical args?

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for your contribution! There are a couple of thinks to merge as well as linting, then we are ready to merge this in!

**Comment by christianmeier** [comment]
Hi Fred, thank you for your comments. I’ll solve them 😊

BR C

 

From: Federico Ciardi ***@***.***>
Reply to: withfig/autocomplete ***@***.***>
Date: Thursday, 27 January 2022 at 18:06
To: withfig/autocomplete ***@***.***>
Cc: christianmeier ***@***.***>, Mention ***@***.***>
Subject: Re: [withfig/autocomplete] feat: add completion spec for tmuxinator (PR #881)

 

@fedeci requested changes on this pull request.

Thanks for your contribution! There are a couple of thinks to merge as well as linting, then we are ready to merge this in!

In src/tmuxinator.ts:
> +    {
 
+      name: "local",
 
+      description:
 
+        "Start a tmux session using ./.tmuxinator.y[a]ml",
 
+    },
 
We could also add file suggestion for this arg, see the filepaths generator sugar: #794

In src/tmuxinator.ts:
> +      name: "start",
 
+      description:
 
+        "Start a tmux session using a project's name (with an optional [ALIAS] for project reuse) or a path to a project config file",
 
+      args: [{
 
+        name: "PROJECT",
 
+        description: "Project name 1",
 
+        isOptional: true,
 
+      },
 
+      {
 
+        name: "ARGS",
 
+        description: "Argument",
 
+        isOptional: true,
 
+      },
 
+      ],
 
There are some options missing for the start command https://github.com/tmuxinator/tmuxinator#starting-a-session, could you add them please?

In src/tmuxinator.ts:
> +const completionSpec: Fig.Spec = {
 
+  name: "tmuxinator",
 
+  description: "Tmuxinator is a ruby gem that allows you to easily manage tmux sessions by using yaml files to describe the layout of a tmux session, and open up that session with a single command",
 
+  subcommands: [
 
+    {
 
+      name: "commands",
 
+      description:
 
+        "Lists commands available in tmuxinator",
 
+    },
 
+    {
 
+      name: "completions",
 
+      description:
 
+        "Used for shell completion",
 
+    },
 
+    {
 
+      name: "copy",
 
There are other subcommands missing aliases, could you add them?

⬇️ Suggested change 
-      name: "copy",
 
+      name: ["copy", "c", "cp"],
 
In src/tmuxinator.ts:
> +      args: [{
 
+        name: "EXISTING",
 
+        description: "The existing project",
 
+        isOptional: true,
 
+      },
 
+      {
 
+        name: "NEW",
 
+        description: "Name new project",
 
+        isOptional: true,
 
+      },
 
+      ],
 
I am not sure whether these are optional 🤔, what happens if you do not provide them?

In src/tmuxinator.ts:
> +      args: [{
 
+        name: "PROJECT1",
 
+        description: "Project name 1",
 
+        isOptional: true,
 
+      },
 
+      {
 
+        name: "PROJECT1",
 
+        description: "Project name 1",
 
+        isOptional: true,
 
+      },
 
+      ],
 
Why this takes two identical args?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by brendanfalk** [comment]
Closing as someone has already merged this

**Pull request closed without merging**